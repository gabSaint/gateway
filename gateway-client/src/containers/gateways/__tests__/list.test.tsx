import React from "react";
import { ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import renderer from "react-test-renderer";

import ListGateways from "../list";

describe("ListGateways", () => {
  let wrapperListGateways: ShallowWrapper | ReactWrapper;

  beforeEach(() => {
    wrapperListGateways = shallow(<ListGateways />);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<ListGateways />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Header", () => {
    it("renders title", () => {
      const title = wrapperListGateways.find("h2");
      expect(title).toHaveLength(1);
    });

    it("renders create new gateway button", () => {
      const newGatewayButton = wrapperListGateways.find(
        "[data-test='button-new-gateway']"
      );
      expect(newGatewayButton).toHaveLength(1);
    });
  });

  describe("Table", () => {
    it("renders Table", () => {
      const table = wrapperListGateways.find("Table");
      expect(table).toHaveLength(1);
    });

    it("renders 5 td foreach tr in body", () => {
      const tableBodyRows = wrapperListGateways.find("tbody>tr");
      const tableBodyColumns = wrapperListGateways.find("tbody>tr>td");
      expect(tableBodyColumns.length).toBe(5 * tableBodyRows.length);
    });
  });

  describe("Buttons", () => {
    let tablebodyLastcolumns;

    beforeEach(() => {
      tablebodyLastcolumns = wrapperListGateways.find("td:last-child");
    });

    it("renders view buttons for each gateway", () => {
      const viewButtons = wrapperListGateways.find("[data-test='button-view']");
      expect(viewButtons).toHaveLength(tablebodyLastcolumns.length);
    });

    it("renders edit buttons for each gateway", () => {
      const editButtons = wrapperListGateways.find("[data-test='button-edit']");
      expect(editButtons).toHaveLength(tablebodyLastcolumns.length);
    });
  });
});
