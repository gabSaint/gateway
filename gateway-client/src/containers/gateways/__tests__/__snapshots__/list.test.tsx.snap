// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGateways renders correctly 1`] = `
Array [
  <h2>
    Gateways
  </h2>,
  <table
    className="ui definition table"
  >
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          Serial number
        </th>
        <th>
          Name
        </th>
        <th>
          Ipv4 address
        </th>
        <th>
          Operations
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
  <div>
    <button
      data-test="button-new-gateway"
    >
      New Gateway
    </button>
  </div>,
]
`;
